### ISSUE TITLE
clientv3.New() hangs when auth enabled and nothing is listening

### ISSUE THREAD

**Issue by wjackson** [coreos/etcd#7747]
The following code hangs forever. If you comment out the `Username` and `Password` lines it fails after one second and correctly reports "connection refused" as expected.
```package main

import (
        "time"

        "github.com/coreos/etcd/clientv3"
)

func main() {
        _, err := clientv3.New(
                clientv3.Config{
                        Endpoints:   []string{"localhost:12345"}, // nothing is listening here
                        DialTimeout: 1 * time.Second,
                        Username:    "someone",
                        Password:    "special",
                },
        )
        if err != nil {
                panic(err)
        }
}
```

Another way to observe the bug:
```
# hangs forever
ETCDCTL_API=3 /usr/local/etcd-v3.1.5-linux-amd64/etcdctl --endpoints=localhost:12345 --user=someone:special get /foo
```
If you remove the `--user` flag it exists after 2 seconds and reports "timed out when dialing".

This bug came up because we turned on auth in our cluster and now go clients hang when one or more of the endpoints are down. The desired behavior is that they will use one of the still working endpoints or produce a timeout error if no endpoints are up.

**Comment by heyitsanthony** [comment]
Likely duplicates #7627 (fix merged into 3.1 as 65382175286473f716acf51f751b43ecdb770a48). Will be in 3.1.6.

**Comment by gyuho** [comment]
Closing. Please try https://github.com/coreos/etcd/releases/tag/v3.1.6
